setup.py

```diff
@@ -1,75 +1,91 @@
-from setuptools import setup, find_packages
-import os
-import re
-import sys
 
-v = open(os.path.join(os.path.dirname(__file__), 'mako', '__init__.py'))
-VERSION = re.compile(r".*__version__ = '(.*?)'", re.S).match(v.read()).group(1)
-v.close()
+from sys import version_info as v
+if any([v < (2, 6), (3,) < v < (3, 3)]):
+    raise Exception("Unsupported Python version %d.%d. Requires Python >= 2.6 "
+                    "or >= 3.3." % v[:2])
 
-readme = open(os.path.join(os.path.dirname(__file__), 'README.rst')).read()
+from setuptools import setup, find_packages
+from os.path import abspath, dirname, join
 
-if sys.version_info < (2, 6):
-    raise Exception("Mako requires Python 2.6 or higher.")
+here = abspath(dirname(__file__))
 
-markupsafe_installs = (
-            sys.version_info >= (2, 6) and sys.version_info < (3, 0)
-        ) or sys.version_info >= (3, 3)
+with open(join(here, 'README.rst')) as r:
+    readme = r.read()
 
-install_requires = []
+tests_require = ['nose >= 0.11']
+if v < (3,):
+    tests_require.append('mock')
 
-if markupsafe_installs:
-    install_requires.append('MarkupSafe>=0.9.2')
+setup(
+    name='Mako',
+    use_scm_version={
+        'version_scheme': 'guess-next-dev',
+        'local_scheme': 'dirty-tag',
+        'write_to': 'mako/_version.py'
+    },
 
-try:
-    import argparse
-except ImportError:
-    install_requires.append('argparse')
+    setup_requires=['setuptools-scm>1.5.4'],
```

Comment:
I'm not sure I can answer that. Mako already has quite a list of dependencies (I think some are missing).
setuptools_scm is a pypa project (like pip and setuptools) - there are no new packages installed when using setup_requires.